rosys/vision/mjpeg_camera/mjpeg_device.py

```diff
@@ -36,50 +35,71 @@ async def restart_capture(self) -> None:
                             self.log.error('could not connect to %s (credentials: %s): %s %s',
                                            self.url, self.authentication, response.status_code, response.reason_phrase)
                             return
-                        buffer = BytesIO()
+
+                        buffer_size = 16 * 1024 * 1024
+                        buffer = bytearray(buffer_size)
+                        buffer_view = memoryview(buffer)
+                        buffer_end = 0
                         header = None
-                        pos = 0
+
+                        byte_search_pos = 0
+
                         try:
                             async for chunk in response.aiter_bytes():
-                                buffer.write(chunk)
+                                chunk_len = len(chunk)
+
+                                if buffer_end + chunk_len > buffer_size:
+                                    self.log.warning('Buffer overflow, resetting buffer')
+                                    buffer_end = 0
+                                    header = None
+                                    continue
+                                buffer_view[buffer_end:buffer_end + chunk_len] = chunk
+                                buffer_end += chunk_len
+
                                 while True:
                                     if header is None:
-                                        header_pos = buffer.getvalue().find(b'\xff\xd8', pos)
+                                        header_pos = buffer.find(b'\xff\xd8', byte_search_pos, buffer_end)
                                         if header_pos == -1:
-                                            pos = max(0, buffer.tell() - 1)
                                             break
-                                        pos = header_pos + 2
+                                        byte_search_pos = header_pos + 2
                                         header = header_pos
                                     else:
-                                        footer_pos = buffer.getvalue().find(b'\xff\xd9', pos)
+                                        footer_pos = buffer.find(b'\xff\xd9', byte_search_pos, buffer_end)
                                         if footer_pos == -1:
-                                            pos = max(0, buffer.tell() - 1)
                                             break
-                                        image_data = buffer.getvalue()[header:footer_pos + 2]
-                                        yield remove_exif(image_data)
-                                        buffer = BytesIO(buffer.getvalue()[footer_pos + 2:])
-                                        pos = 0
+                                        image_data = buffer[header:footer_pos + 2]
+                                        yield image_data
+
+                                        buffer_view[:buffer_end - (footer_pos + 2)
```

Comment:
The line wraps a bit weirdly. What about this alternative:
```py
buffer_view[:buffer_end - (footer_pos + 2)] = \
    buffer_view[footer_pos + 2:buffer_end]
```
One way or the other, it's pretty hard to read. Should we introduce a helper variable like `shift = footer_pos + 2` to underline the purpose of this code block?
```py
shift = footer_pos + 2
buffer_view[:buffer_end - shift] = buffer_view[shift:buffer_end]
header = None
buffer_end -= shift
byte_search_pos = 0
```